### PR TITLE
Add aoi / extend / bookmark component and service

### LINKED ISSUE #148: User preference : favorite AOI (draw, shp, geosjon,kml/kmz,wkt) with localstorage
Implement a new component which list favorite AOI of the user,
In the list I can :
-View AOI (close the modal and zoom to)
-Remove AOI
The add AOI function should be implemented in this component and use when user add AOI (optional in import)
We work with the local storage in the first time, but think the dev with a service to plug arlas-preference api.

### PR DESCRIPTION
**Pull request by sebbousquet**
Fix #148

### CODE REVIEW

File: src/app/components/bookmark-menu/bookmark-menu.component.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<div class="bookmark-shortcut" >
+  <mat-chip-list>
+    <mat-basic-chip [hidden]="!isBookmarkOpen" *ngFor="let bookmark of topBookmarks;let index = index" class="bookmark-chip bookmark-view" (click)="viewBookmark(bookmark.id)">
+      <span class="bookmark-name">#{{index+1}} - {{ bookmark.name}}</span>
+    </mat-basic-chip>
+    <mat-basic-chip [hidden]="!isBookmarkOpen" class="bookmark-chip bookmark-action-left" style="margin-left:5px" matTooltip="Add a dataset" (click)="openDialogAdd()">
```
Comment (MohamedHamouGisaia) [inline src/app/components/bookmark-menu/bookmark-menu.component.html]: we can pipe 'Add dataset' with `translate`

File: src/app/components/bookmark-menu/bookmark-menu.component.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      <mat-icon aria-label="Add a dataset">add</mat-icon>
+    </mat-basic-chip>
+    <mat-basic-chip [hidden]="!isBookmarkOpen" class="bookmark-chip bookmark-action-right" matTooltip="Manage datasets" (click)="openDialog()">
```
Comment (MohamedHamouGisaia) [inline src/app/components/bookmark-menu/bookmark-menu.component.html]: piping 'Manage dataset' with `translate` too

**Comment by MohamedHamouGisaia** [comment]
I noticed that `bookmark.component.ts`, `extend.component.ts` and `aoi.component.ts` have 3 attributes in common and the same implementation of `selectBookmark` , `selectAoi` (same implementation of adding and removing elements from `this.itemsCheck`.

Maybe we can add A generic class to mutualize them?

Other than that, everythinh LGTM

File: src/app/components/bookmark-menu/bookmark-menu.component.html (hunk 2 above)
Comment (sebbousquet) [reply src/app/components/bookmark-menu/bookmark-menu.component.html]: Done

File: src/app/components/bookmark-menu/bookmark-menu.component.html (hunk 1 above)
Comment (sebbousquet) [reply src/app/components/bookmark-menu/bookmark-menu.component.html]: Done

**Comment by sebbousquet** [comment]
> I noticed that `bookmark.component.ts`, `extend.component.ts` and `aoi.component.ts` have 3 attributes in common and the same implementation of `selectBookmark` , `selectAoi` (same implementation of adding and removing elements from `this.itemsCheck`.
> 
> Maybe we can add A generic class to mutualize them?
> 

It's not exactly the same implementation. It's a need to separate each components logic even if, for now, we have a little code duplication

**Pull request merged**